python/rikai/spark/functions/vision.py

```diff
@@ -133,3 +142,43 @@ def video_to_images(
         )
         for idx, img in enumerate(video_iterator)
     ]
+
+
+@udf(returnType=ImageType())
+def spectrogram_image(video, size: int = 224) -> Image:
+    import ffmpeg
+
+    """Applied ffmpeg filter to generate spectrogram.
+
+    Parameters
+    ----------
+    video : Video
+        A video object, either YouTubeVideo or VideoStream.
+    size : Int
+        Sets resolution of frequency, time spectrogram image.
+
+    Return
+    ------
+    Image
+        Return an Image of the audio spectrogram.
+    """
+    assert isinstance(video, YouTubeVideo) or isinstance(
```

Comment:
`isinstance(video, (YouTubeVideo, VideoStream))` should work

Reply:
got it